diff -ruN rr232x_org/rr232x-linux-src-v1.10/inc/linux/Makefile.def rr232x_new/rr232x-linux-src-v1.10/inc/linux/Makefile.def
--- rr232x_org/rr232x-linux-src-v1.10/inc/linux/Makefile.def	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/inc/linux/Makefile.def	2014-03-08 17:33:33.274680855 +0100
@@ -74,19 +74,31 @@
 KERNELDIR := /lib/modules/$(shell uname -r)/build
 endif
 
-KERNEL_VER := 2.$(shell expr `grep LINUX_VERSION_CODE $(KERNELDIR)/include/linux/version.h | cut -d\  -f3` / 256 % 256)
+KERNEL_VER := $(shell uname -r | cut -f1-2 -d.)
 
 ifeq ($(KERNEL_VER),)
-$(error Cannot find kernel version. Check $(KERNELDIR)/include/linux/version.h.)
+$(error Cannot find kernel version. Check $(KERNELDIR)/include/generated/uapi/linux/version.h.)
 endif
 
-ifneq ($(KERNEL_VER), 2.6)
-ifneq ($(KERNEL_VER), 2.4)
-$(error Only kernel 2.4/2.6 is supported but you use $(KERNEL_VER))
+ifneq ($(KERNEL_VER), 3.4)
+ifneq ($(KERNEL_VER), 3.6)
+ifneq ($(KERNEL_VER), 3.8)
+ifneq ($(KERNEL_VER), 3.10)
+ifneq ($(KERNEL_VER), 3.11)
+ifneq ($(KERNEL_VER), 3.13)
+ifneq ($(KERNEL_VER), 3.14)
+ifneq ($(KERNEL_VER), 3.15)
+$(error Only kernel 3.4/3.6/3.8/3.10/3.11/3.13/3.13/3.14/3.15 is supported but you use $(KERNEL_VER))
+endif
+endif
+endif
+endif
+endif
+endif
 endif
 endif
 
-ifeq ($(KERNEL_VER), 2.6)
+ifeq ($(KERNEL_VER), $(filter $(KERNEL_VER),3.4 3.6 3.8 3.10 3.11 3.13 3.14 3.15))
 
 TARGET := $(TARGETNAME).ko
 
diff -ruN rr232x_org/rr232x-linux-src-v1.10/osm/linux/install.sh rr232x_new/rr232x-linux-src-v1.10/osm/linux/install.sh
--- rr232x_org/rr232x-linux-src-v1.10/osm/linux/install.sh	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/osm/linux/install.sh	2014-03-08 17:36:41.690487903 +0100
@@ -6,11 +6,35 @@
 PWD=`pwd`
 
 case ${KERNEL_VER} in
-	2.4 )
-	OBJ=o
-	MODVER=`modinfo -f%{kernel_version} ${PWD}/${TARGETNAME}.${OBJ}`
+	3.4 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.6 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.8 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.10 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.11 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.13 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
+	;;
+	3.14 )
+	OBJ=ko
+	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
 	;;
-	2.6 )
+	3.15 )
 	OBJ=ko
 	MODVER=`modinfo -F vermagic ${PWD}/${TARGETNAME}.${OBJ} | cut -d' ' -f1`
 	;;
diff -ruN rr232x_org/rr232x-linux-src-v1.10/osm/linux/os_linux.c rr232x_new/rr232x-linux-src-v1.10/osm/linux/os_linux.c
--- rr232x_org/rr232x-linux-src-v1.10/osm/linux/os_linux.c	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/osm/linux/os_linux.c	2014-03-08 17:31:12.688332528 +0100
@@ -198,7 +198,7 @@
 
 	if (page)
 		return (PageHighMem(page)?
-				(char *)kmap_atomic(page, HPT_KMAP_TYPE) :
+				(char *)kmap_atomic(page) :
 				(char *)page_address(page))
 			+ (psg->addr.bus & 0xffffffff);
 	else
@@ -208,7 +208,7 @@
 void os_kunmap_sgptr(void *ptr)
 {
 	if ((HPT_UPTR)ptr >= (HPT_UPTR)high_memory)
-		kunmap_atomic(ptr, HPT_KMAP_TYPE);
+		kunmap_atomic(ptr);
 }
 #else 
 void *os_kmap_sgptr(PSG psg) { return psg->addr._logical; }
@@ -260,7 +260,7 @@
 					struct block_device *bdev = bdget(MKDEV(major[i], minor));
 					if (bdev &&
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,28)
-						blkdev_get(bdev, FMODE_READ)
+						blkdev_get(bdev, FMODE_READ, NULL)
 #else 
 						blkdev_get(bdev, FMODE_READ, 0 __BDEV_RAW)
 #endif
diff -ruN rr232x_org/rr232x-linux-src-v1.10/osm/linux/osm_linux.c rr232x_new/rr232x-linux-src-v1.10/osm/linux/osm_linux.c
--- rr232x_org/rr232x-linux-src-v1.10/osm/linux/osm_linux.c	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/osm/linux/osm_linux.c	2014-03-08 17:31:12.689332537 +0100
@@ -446,17 +446,17 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,23)
 	struct scatterlist *sg;
 	sg = scsi_sglist(cmd);
-	*pbuf = kmap_atomic(HPT_SG_PAGE(sg), HPT_KMAP_TYPE) + sg->offset;
+	*pbuf = kmap_atomic(HPT_SG_PAGE(sg)) + sg->offset;
 	buflen = sg->length;
 #else 
 
 	if (cmd->use_sg) {
 		struct scatterlist *sg = (struct scatterlist *) cmd->request_buffer;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
-		*pbuf = kmap_atomic(HPT_SG_PAGE(sg), HPT_KMAP_TYPE) + sg->offset;
+		*pbuf = kmap_atomic(HPT_SG_PAGE(sg)) + sg->offset;
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,18)
 		if (sg->page)
-			sg->address = kmap_atomic(sg->page, HPT_KMAP_TYPE) + sg->offset;
+			sg->address = kmap_atomic(sg->page) + sg->offset;
 		*pbuf = sg->address;
 #else 
 		*pbuf = sg->address;
@@ -476,16 +476,16 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,23)
 	struct scatterlist *sg;
 	sg = scsi_sglist(cmd);
-	kunmap_atomic((char *)buf - sg->offset, HPT_KMAP_TYPE);
+	kunmap_atomic((char *)buf - sg->offset);
 #else 
 
 	if (cmd->use_sg) {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
-		kunmap_atomic((char *)buf - ((struct scatterlist *)cmd->request_buffer)->offset, HPT_KMAP_TYPE);
+		kunmap_atomic((char *)buf - ((struct scatterlist *)cmd->request_buffer)->offset);
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,18)
 		struct scatterlist *sg = (struct scatterlist *) cmd->request_buffer;
 		if (sg->page)
-			kunmap_atomic((char *)buf - sg->offset, HPT_KMAP_TYPE);
+			kunmap_atomic((char *)buf - sg->offset);
 #endif
 	}
 
@@ -874,7 +874,7 @@
 	}
 }
 
-static int hpt_queuecommand (Scsi_Cmnd * SCpnt, void (*done) (Scsi_Cmnd *))
+static int hpt_queuecommand_lck (Scsi_Cmnd * SCpnt, void (*done) (Scsi_Cmnd *))
 {
 	struct Scsi_Host *phost = sc_host(SCpnt);
 	PVBUS_EXT vbus_ext = get_vbus_ext(phost);
@@ -1408,6 +1408,13 @@
 	return 0;
 }
 
+
+#ifdef DEF_SCSI_QCMD
+DEF_SCSI_QCMD(hpt_queuecommand)
+#else
+#define hpt_queuecommand hpt_queuecommand_lck
+#endif
+
 static int hpt_reset (Scsi_Cmnd *SCpnt)
 {
 	PVBUS_EXT vbus_ext = get_vbus_ext(sc_host(SCpnt));
@@ -2089,7 +2096,7 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0) /* 2.4.x */
 	use_new_eh_code:         1,
 	proc_name:               driver_name,
-	proc_info:               hpt_proc_info24,
+	show_info:               hpt_proc_info24,
 	select_queue_depths:     NULL,
 	max_sectors:             128,
 
@@ -2098,7 +2105,7 @@
 	#endif
 #else /* 2.6.x */
 	proc_name:               driver_name,
-	proc_info:               hpt_proc_info26,
+	show_info:               hpt_proc_info26,
 	max_sectors:             128,
 #endif
 	this_id:                 -1
diff -ruN rr232x_org/rr232x-linux-src-v1.10/osm/linux/osm_linux.h rr232x_new/rr232x-linux-src-v1.10/osm/linux/osm_linux.h
--- rr232x_org/rr232x-linux-src-v1.10/osm/linux/osm_linux.h	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/osm/linux/osm_linux.h	2014-03-08 17:31:12.690332547 +0100
@@ -8,11 +8,7 @@
 
 /* system headers */
 
-#ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
-#endif
-
-#include <linux/version.h>
+#include </usr/include/linux/version.h>
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)) && defined(MODVERSIONS)
 #include <linux/modversions.h>
@@ -174,10 +170,8 @@
 #endif
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
-#define HPT_KMAP_TYPE KM_BIO_SRC_IRQ
 #define HPT_FIND_PCI_DEVICE pci_get_device
 #else 
-#define HPT_KMAP_TYPE KM_BH_IRQ
 #define HPT_FIND_PCI_DEVICE pci_find_device
 #endif
 
diff -ruN rr232x_org/rr232x-linux-src-v1.10/osm/linux/patch.sh rr232x_new/rr232x-linux-src-v1.10/osm/linux/patch.sh
--- rr232x_org/rr232x-linux-src-v1.10/osm/linux/patch.sh	2009-07-16 03:28:28.000000000 +0200
+++ rr232x_new/rr232x-linux-src-v1.10/osm/linux/patch.sh	2014-03-08 17:31:12.690332547 +0100
@@ -7,12 +7,6 @@
 if test "${TARGETMODS-set}" = set; then echo "TARGETMODS is not set"; exit 1; fi
 if test "${TARGETOBJS-set}" = set; then echo "TARGETOBJS is not set"; exit 1; fi
 
-#check to see if it is really a src dir
-if [ ! -d ${KERNELDIR}/Documentation ] ; then
-	echo "The target directory ${KERNELDIR} is not a full kernel source tree."
-	exit 1
-fi
-
 ARCH=$( uname -m )
 PRODUCTNAME=$( ( cd .. ; pwd ) | sed "s/\//\n/g" | tail -n1 )
 UTARGETNAME=$( echo $TARGETNAME | tr "[:lower:]" "[:upper:]" )
diff -ruN rr232x_org/rr232x-linux-src-v1.10/product/rr232x/linux/.build/Makefile rr232x_new/rr232x-linux-src-v1.10/product/rr232x/linux/.build/Makefile
--- rr232x_org/rr232x-linux-src-v1.10/product/rr232x/linux/.build/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ rr232x_new/rr232x-linux-src-v1.10/product/rr232x/linux/.build/Makefile	2014-03-08 17:31:12.691332556 +0100
@@ -0,0 +1,21 @@
+# auto generated makefile
+HPT_ROOT := /root/sources/rr232x-linux-src-v1.10_up
+HPT_REGPARM := $(shell echo "$(CFLAGS) $(KBUILD_CFLAGS)" | awk '{ match($$0,"-mregparm="); if (RSTART>0) print substr($$0,RSTART+10,1);}')
+HPT_REGPARM := $(if $(HPT_REGPARM),$(HPT_REGPARM),0)
+HPT_LIB := /root/sources/rr232x-linux-src-v1.10_up/lib/linux/free-x86_64-regparm$(HPT_REGPARM)
+TARGETNAME := rr232x
+TARGETOBJS := os_linux.o osm_linux.o div64.o hptinfo.o config.o
+TARGETMODS := him_rr232x.o ldm_raid50.o partition.o raid0.o raid1.o raid5.o jbod.o
+
+obj-m := $(TARGETNAME).o
+
+$(TARGETNAME)-objs := $(TARGETOBJS) $(TARGETMODS)
+
+EXTRA_CFLAGS := -DLINUX -D_LINUX_ -D__KERNEL__=1 -DSUPPORT_ARRAY -DDBG=0 -DBITS_PER_LONG=64 -DMODVERSIONS -DMODULE -Idrivers/scsi -I$(HPT_ROOT)/inc/linux -I$(HPT_ROOT)/inc -I$(HPT_ROOT)/osm/linux
+
+$(addprefix $(obj)/,$(TARGETOBJS:.o=.c)): FORCE
+	@{ test -f /root/sources/rr232x-linux-src-v1.10_up/product/rr232x/linux/$(notdir $@) && cp -f /root/sources/rr232x-linux-src-v1.10_up/product/rr232x/linux/$(notdir $@) $@; } ||\
+	{ test -f $(HPT_ROOT)/osm/linux/$(notdir $@) && cp -f $(HPT_ROOT)/osm/linux/$(notdir $@) $@; }
+
+$(addprefix $(obj)/,$(TARGETMODS)): $(obj)/%.o: $(HPT_LIB)/%.o
+	@cp -f $< $@
