diff -rupN obe-vod_org/configure obe-vod_new/configure
--- obe-vod_org/configure	2013-09-27 15:07:25.248944725 +0200
+++ obe-vod_new/configure	2013-09-27 15:08:22.368942374 +0200
@@ -934,18 +934,12 @@ if [ "$gpac" = "auto" ] ; then
         GPAC_LIBS="$GPAC_LIBS -lwinmm"
     fi
     if cc_check gpac/isomedia.h "$GPAC_LIBS" ; then
-        if cc_check gpac/isomedia.h "$GPAC_LIBS" "gf_isom_set_pixel_aspect_ratio(0,0,0,0,0);" ; then
-            gpac="yes"
-        else
-            echo "Warning: gpac is too old, update to 2007-06-21 UTC or later"
-        fi
+        gpac="yes"
     fi
 fi
 if [ "$gpac" = "yes" ] ; then
     define HAVE_GPAC
-    if cc_check gpac/isomedia.h "-Werror $GPAC_LIBS" "void *p; p = gf_malloc(1); gf_free(p);" ; then
-        define HAVE_GF_MALLOC
-    fi
+    define HAVE_GF_MALLOC
     LDFLAGSCLI="$GPAC_LIBS $LDFLAGSCLI"
 fi
 
Binary files obe-vod_org/.git/index and obe-vod_new/.git/index differ
