--- fffffff/CMakeLists.txt	2025-04-09 00:34:33.103935540 +0200
+++ fffff/CMakeLists.txt	2025-04-09 00:35:29.443171609 +0200
@@ -26,13 +26,8 @@
 	target_compile_options(${LIBRARY_NAME} PRIVATE /bigobj /fp:fast)
 else()
 	# If you change this, make sure to update the corresponding line in the pybind CMakeLists
-	if (MARCH_NATIVE)
-		target_compile_options(${LIBRARY_NAME} PRIVATE
-			-march=native -Wall -Werror -fPIC -Wno-ignored-optimization-argument)
-	else()
-		target_compile_options(${LIBRARY_NAME} PRIVATE
-			-Wall -Werror -fPIC)
-	endif()
+	target_compile_options(${LIBRARY_NAME} PRIVATE
+		-Wall -fPIC)
 	if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 		target_compile_options(${LIBRARY_NAME} PRIVATE
 				-Wno-maybe-uninitialized)
